content/intro-to-storybook/vue/es/data.md

```diff
@@ -0,0 +1,232 @@
+
+```html
+
+<!--src/App.vue-->
+<template>
+  <div id="app">
+    <task-list />
+  </div>
+</template>
+
+<script>
+  import store from './store';
+  import TaskList from './components/TaskList.vue';
+
+  export default {
+    name: 'app',
+    store,
+    components: {
+      TaskList,
+    },
+  };
+</script>
+<style>
+  @import './index.css';
+</style>
+```
+
+Luego se cambiará el componente `TaskList` para leer los datos del store. Pero primero, pasemos nuestra versión del componente existente al archivo `src/components/PureTaskList.vue` (renombrar el componente a `pure-task-list`) que luego se incluirá en un contenedor.
+
+En `src/components/PureTaskList.vue`:
+
+```html
+<!--src/components/PureTaskList.vue-->
+<template>
+<!--same content as before-->
+</template>
+<script>
+import Task from "./Task";
+export default {
+  name: "pure-task-list",
+  ...
+}
+```
+
+En `src/components/TaskList.vue`:
+
+```html
+
+<!--src/components/TaskList.vue`-->
+<template>
+  <div>
+    <pure-task-list :tasks="tasks" @archiveTask="archiveTask" @pinTask="pinTask" />
+  </div>
+</template>
+
+<script>
+  import PureTaskList from './PureTaskList';
+  import { mapState, mapActions } from 'vuex';
+
+  export default {
+    name: 'task-list',
+    components: {
+      PureTaskList,
+    },
+    methods: {
+      ...mapActions(['archiveTask', 'pinTask']),
+    },
+    computed: {
+      ...mapState(['tasks']),
+    },
+  };
+</script>
+```
+
+La razón para mantener separada la versión de la `TaskList` es porque es más fácil de probar y aislar. Como no depende de la presencia de un store, es mucho más fácil tratar desde una perspectiva de prueba. Cambiemos el nombre de `src/components/TaskList.stories.js` a`src/components/PureTaskList.stories.js`, con esto garantizamos que nuestras stories usen la versión actual:
+
+```javascript
+
+//src/components/PureTaskList.stories.js
+import PureTaskList from './PureTaskList';
+import { taskData, actionsData } from './Task.stories';
+
+const paddedList = () => {
+  return {
+    template: '<div style="padding: 3rem;"><story/></div>',
+  };
+};
+export default {
+  title: 'TaskList',
+  excludeStories: /.*Data$/,
+  decorators: [paddedList],
+};
+
+export const defaultTasksData = [
+  { ...taskData, id: '1', title: 'Task 1' },
+  { ...taskData, id: '2', title: 'Task 2' },
+  { ...taskData, id: '3', title: 'Task 3' },
+  { ...taskData, id: '4', title: 'Task 4' },
+  { ...taskData, id: '5', title: 'Task 5' },
+  { ...taskData, id: '6', title: 'Task 6' },
+];
+export const withPinnedTasksData = [
+  ...defaultTasksData.slice(0, 5),
+  { id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
+];
+
+export const Default = () => ({
+  components: { PureTaskList },
+  template: `<pure-task-list :tasks="tasks" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
+  props: {
+    tasks: {
+      default: () => (withPinnedTasksData)
```

Comment:
@IvanSotelo like in the section before, in the composite component the extra parenthesis are not needed, to avoid introducing back issues with prettier, same as with the `WithPinnedTasks `